internal/storage/storageutil/client.go

```diff
@@ -107,3 +108,11 @@ func CreateTokenSource(storageClientConfig *StorageClientConfig) (tokenSrc oauth
 		return oauth2.StaticTokenSource(&oauth2.Token{}), nil
 	}
 }
+
+// StripScheme strips the scheme from a host if it contains.
+func StripScheme(url string) string {
```

Comment:
`you can use url.Host instead of this function ?` - Yeah, I tried this first, but that is not working.

Parsing of endpoint to url doesn't work if someone gives endpoint without scheme. 

```
// Parse parses a raw url into a [URL] structure.
//
// The url may be relative (a path, without a host) or absolute
// (starting with a scheme). Trying to parse a hostname and path
// without a scheme is invalid but may not necessarily return an
// error, due to parsing ambiguities.
func Parse(rawURL string) (*URL, error) {
```

If you will pass - custom-endpoint=localhost:999, then you will get url.Host as empty.
Also, we don't return any error if some one passes without scheme.

Reply:
Discussed offline, so resolving this.